pkg/datapath/linux/probes/probes.go

```diff
@@ -201,85 +208,96 @@ func (p *ProbeManager) SystemKernelHz() (int, error) {
 // returns an error when parameters required by Cilium are not enabled. It logs
 // warnings when optional parameters are not enabled.
 func (p *ProbeManager) SystemConfigProbes() error {
-	config := p.features.SystemConfig
-
 	if !p.KernelConfigAvailable() {
 		return ErrKernelConfigNotFound
 	}
-
-	// Required
-	if !config.ConfigBpf.Enabled() {
-		return fmt.Errorf("CONFIG_BPF kernel parameter is required")
-	}
-
-	if !config.ConfigBpfSyscall.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_BPF_SYSCALL kernel parameter is required")
-	}
-	if !config.ConfigNetSchIngress.Enabled() && !config.ConfigNetSchIngress.Module() {
-		return fmt.Errorf(
-			"CONFIG_NET_SCH_INGRESS kernel parameter (or module) is required")
-	}
-	if !config.ConfigNetClsBpf.Enabled() && !config.ConfigNetClsBpf.Module() {
-		return fmt.Errorf(
-			"CONFIG_NET_CLS_BPF kernel parameter (or module) is required")
-	}
-	if !config.ConfigNetClsAct.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_NET_CLS_ACT kernel parameter is required")
-	}
-	if !config.ConfigBpfJit.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_BPF_JIT kernel parameter is required")
-	}
-	if !config.ConfigHaveEbpfJit.Enabled() {
-		return fmt.Errorf(
-			"CONFIG_HAVE_EBPF_JIT kernel parameter is required")
-	}
-	// Optional
-	if !config.ConfigCgroupBpf.Enabled() {
-		log.Warning(
-			"CONFIG_CGROUP_BPF optional kernel parameter is not in kernel configuration")
-	}
-	if !config.ConfigLwtunnelBpf.Enabled() {
-		log.Warning(
-			"CONFIG_LWTUNNEL_BPF optional kernel parameter is not in kernel configuration")
+	requiredParams := p.GetRequiredConfig()
+	for param, kernelOption := range requiredParams {
+		if !kernelOption.Enabled && kernelOption.CanBeModule == true {
+			return fmt.Errorf("%s kernel parameter or module is required", param)
+		} else {
+			return fmt.Errorf("%s kernel parameter is required", param)
+		}
 	}
-	if !config.ConfigBpfEvents.Enabled() {
-		log.Warning(
-			"CONFIG_BPF_EVENTS optional kernel parameter is not in kernel configuration")
+	optionalParams := p.GetOptionalConfig()
+	for param, kernelOption := range optionalParams {
+		if !kernelOption.Enabled && kernelOption.CanBeModule == true {
+			log.Warningf("%s optional kernel parameter or module is not in kernel", param)
+		} else {
+			log.Warningf("%s optional kernel parameter or module is not in kernel", param)
+		}
```

Comment:
Same logics error, plus you have twice the same message anyway :).
Also need to print the `.Description` to tell users what the parameter is needed for?